algorithms/kernel/svm/svm_train_thunder_impl.i

```diff
@@ -307,29 +341,64 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::SMO
 }
 
 template <typename algorithmFPType, typename ParameterType, CpuType cpu>
-services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::updateGrad(const algorithmFPType * kernelWS,
-                                                                                        const algorithmFPType * deltaalpha, algorithmFPType * grad,
+services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::updateGrad(const NumericTablePtr & kernelWS,
+                                                                                        const algorithmFPType * deltaalpha,
+                                                                                        algorithmFPType * gradBuff, algorithmFPType * grad,
                                                                                         const size_t nVectors, const size_t nWS)
 {
-    char trans = 'N';
-    DAAL_INT m = nVectors;
-    DAAL_INT n = nWS;
-    algorithmFPType alpha(1.0);
-    DAAL_INT ldA = m;
+    SafeStatus safeStat;
+    const size_t nBlocksWS = nWS / _blockSizeWS;
+
+    size_t blockSize   = 256;
+    size_t nBlocksGrad = (nVectors / blockSize) + !!(nVectors % blockSize);
+
     DAAL_INT incX(1);
-    algorithmFPType beta(1.0);
     DAAL_INT incY(1);
 
-    Blas<algorithmFPType, cpu>::xgemv(&trans, &m, &n, &alpha, kernelWS, &ldA, deltaalpha, &incX, &beta, grad, &incY);
+    daal::threader_for(nBlocksWS, nBlocksWS, [&](const size_t iBlock) {
+        const size_t startRowWS           = iBlock * _blockSizeWS;
+        const algorithmFPType deltaalphai = deltaalpha[startRowWS];
+
+        daal::threader_for(nBlocksGrad, nBlocksGrad, [&](const size_t iBlockGrad) {
+            const size_t nRowsInBlockGrad = (iBlockGrad != nBlocksGrad - 1) ? blockSize : nVectors - iBlockGrad * blockSize;
+            const size_t startRowGrad     = iBlockGrad * blockSize;
+            algorithmFPType * gradi       = &gradBuff[nVectors * iBlock + startRowGrad];
+
+            ReadColumns<algorithmFPType, cpu> mtKernel(kernelWS.get(), startRowWS, startRowGrad, nRowsInBlockGrad);
+            DAAL_CHECK_BLOCK_STATUS_THR(mtKernel);
+            const algorithmFPType * kernelBlockI = mtKernel.get();
+            PRAGMA_IVDEP
+            PRAGMA_VECTOR_ALWAYS
+            for (size_t j = 0; j < nRowsInBlockGrad; j++)
+            {
+                gradi[j] = deltaalphai * kernelBlockI[j];
+            }
+
+            for (size_t i = 1; i < _blockSizeWS; i++)
```

Comment:
seems here we could use `gemm` instead of `xxaxpy` if we read bunch of columns instead of one by one,
do you think is it worth a try?